src/components/__tests__/Tasks.test.js

```diff
@@ -0,0 +1,139 @@
+import { render, screen, waitFor } from "@testing-library/react"
+import userEvent from '@testing-library/user-event'
+import { Tasks } from '../Tasks';
+
+const mockItem = {
+    id: 2,
+    isComplete: false,
+    name: 'test-name'
+};
+
+const mockAddItem = jest.fn();
+const mockGetItems = jest.fn();
+const mockDeleteItem = jest.fn();
+const mockUpdateItem = jest.fn();
+
+jest.mock('../../api', () => ({
+    addItem: (item) => mockAddItem(item),
+    getItems: () => mockGetItems(),
+    deleteItem: (id) => mockDeleteItem(id),
+    updateItem: (item) => mockUpdateItem(item),
+}));
+
+beforeEach(() => mockGetItems.mockReturnValue([mockItem]));
+
+describe('Tasks', () => {
+    it('renders correctly', async () => {
+        render(<Tasks />);
+
+        expect(await screen.findByText('Add new task:')).toBeInTheDocument();
+    })
+
+    it('should add an item and reset the new task form after filling the form and clicking add', async () => {
+        render(<Tasks />);
```

Comment:
you will probably need to mock a return value for addItem in here so when it is called you have a mock response from the api

Reply:
Or if it is already mocked then you won't need to (looks like it is on line 5)